src/MOI_wrapper/scaled_psd_cone_bridge.jl

```diff
@@ -3,6 +3,14 @@
 # Use of this source code is governed by an MIT-style license that can be found
 # in the LICENSE.md file or at https://opensource.org/licenses/MIT.
 
+"""
+    struct ScaledPSDCone <: MOI.AbstractVectorSet
+        side_dimension::Int
+    end
+
+Similar to `MOI.ScaledPositiveSemidefiniteConeTriangle` but it the vectorization
+is the lower triangular part column-wise (or the upper triangular part row-wise).
+"""
 struct ScaledPSDCone <: MOI.AbstractVectorSet
```

Comment:
This seems a bit pointless if we're still going to need the bridge...

Now `@constraint(model, X in PSDCone())` is going to need two bridges instead of one...

Reply:
What's the issue of using several bridges ? We should go for simple atomic bridges. If there is an issue about intermediate copies, we can think about fusing the operations or mutating the functions instead of copying them but I'd rather keep the solver wrappers simple and let MOI do the tricky stuff in view of the growing list of solvers.

Reply:
Also, this allows passing scaled PSD constraints to SCS directly while without this PR it would unscale and then scale again

Reply:
There's nothing wrong with multiple bridges. (Except that it'll make https://jump.dev/JuMP.jl/stable/tutorials/conic/ellipse_approx/#Alternative-formulations more verbose.)

But it just seems pointless to rescale the matrix only to then permute it in a different bridge. With the new test system, the overhead for creating and maintaining a bridge is pretty low and this one in particular is well tested.

Perhaps we could have two bridges in SCS?

Reply:
> By the way, the PSD cone is missing from the README

Yes. I just realized that. It's because it needs a bridge and we decided that solvers shouldn't report all sets that they support with bridges :smile: But we could add `SCS.ScaledPSDCone`

Reply:
> But it just seems pointless to rescale the matrix only to then permute it in a different bridge.

I don't see why that's pointless. The bridges are really just matrix multiplication, here we are multiplying by a diagonal and a permutation matrix, that makes sense that it's the product of two different matrices, also it makes it apparent which one is multiplied first since they are not commutative.

> With the new test system, the overhead for creating and maintaining a bridge is pretty low and this one in particular is well tested.

I'd rather not have both. Note that there is also the issue of allocating this scaling vector that was resolved in the MOI bridge (see https://github.com/jump-dev/SCS.jl/issues/228). This was reported in 2019 in https://github.com/jump-dev/SCS.jl/pull/165 but not reported again. That doesn't mean that no user gets any issue, we just gets a very small percentage of issues people encounter as bug report I'd still rather have fewer bridges that are hit by more users.

> But we could add `SCS.ScaledPSDCone`

Yes, I think we should :-P

Reply:
The performance problem was a bug. It just used a poor choice of data structure.

Reply:
I don't see why a `BitSet` was worse than a `Vector{Float64}` (which is what is used now by the bridge). In any case, that still shows that we probably don't want to duplicate the number of equivalent bridges.

Reply:
The `BitSet` problem is separate to the `scale` `Vector{Float64}`. 

We used to do:
```julia
diagidx = BitSet()
for i in 1:d
    push!(diagidx, trimap(i, i))
end
for i in 1:length(output)
    if !(rows[i] in diagidx)
        output[i] *= scaling
    end
end
```

#165 changed it to:
```julia
for i in 1:length(output)
    val = 8 * rows[i] + 1
    if isqrt(val)^2 != val
        output[i] *= scaling
    end
end
```

The bridge now does:
```julia
scale_factor = fill(scale, d)
for i in 1:d
    if MOI.Utilities.is_diagonal_vectorized_index(i)
        scale_factor[i] = 1.0
    end
end
```

Reply:
Yes, the code sample in the middle is the only one that does not allocate, doesn't it ?
Anyway, the issue with keeping the two bridge is that one of them will receive very low coverage which is also an issue, isn't it ?

Reply:
I guess it's not quite my example. Because the matrix is sparse, not dense. 

If you take the example from https://github.com/jump-dev/SCS.jl/issues/228#issuecomment-984293311, there's no performance issue with `SCS#master`. Almost all the time is spent inside SCS. Where it used to take 5 minutes with the `BitSet`.

Reply:
Sure, with that example there is no issue but that does not mean that there is no example with an issue. https://github.com/jump-dev/SCS.jl/pull/165 report an example for which there is an issue.

Reply:
Oh, ok I missed that they are the same sorry. Not sure why allocating vectors would be cheaper than `BitSet`, maybe it was because we were not using `sizehint!` but I guess we don't have to spend time on this if we just agree that it's best not to have two bridges in SCS as then one will get no coverage ?
I don't see why having more lines in the output of the bridge debugging is bad. There is actually two transformations going on, we need to scale and we need to reorder so it's quite intuitive to see both.
The users also sees that if he has the scaled matrices, he can remove one of the two bridges while if you have one bridge that does the two you won't see that so it's loosing some transparency aspect of the bridging.